storage/innobase/fsp/fsp0fsp.cc

```diff
@@ -2864,43 +2922,12 @@ fseg_free_step(
 		fil_block_check_type(*iblock, FIL_PAGE_INODE, mtr);
 	}
 
-	dberr_t err;
-	descr = fseg_get_first_extent(inode, space, mtr, &err);
-
-	if (descr) {
-		/* Free the extent held by the segment */
-		return fseg_free_extent(inode, iblock, space,
-					xdes_get_offset(descr), mtr
-#ifdef BTR_CUR_HASH_ADAPT
-					, ahi
-#endif /* BTR_CUR_HASH_ADAPT */
-					) != DB_SUCCESS;
-	}
-
-	if (err != DB_SUCCESS || space->is_stopping()) {
-		return true;
-	}
-
-	/* Free a frag page */
-	n = fseg_find_last_used_frag_page_slot(inode);
-
-	if (n == ULINT_UNDEFINED) {
-		/* Freeing completed: free the segment inode */
-		fsp_free_seg_inode(space, inode, iblock, mtr);
-		return true;
-	}
-
-	uint32_t page_no = fseg_get_nth_frag_page_no(inode, n);
-
-	if (fseg_free_page_low(inode, iblock, space, page_no, mtr
+	if (fseg_free_step_low(iblock, inode, space, mtr, nullptr, nullptr
 #ifdef BTR_CUR_HASH_ADAPT
 			       , ahi
 #endif /* BTR_CUR_HASH_ADAPT */
-			       ) != DB_SUCCESS) {
-		return true;
-	}
-
-	buf_page_free(space, page_no, mtr);
+                                ))
+         return true;
```

Comment:
I would retain the original formatting here, only removing the ` != DB_SUCCESS` since the function returns `bool` instead of `dberr_t`.